src/prommis/solid_handling/crusher.py

```diff
@@ -0,0 +1,231 @@
+
+Additional Constraints
+----------------------
+
+Crusher adds one additional constraint to calculate the work required to crush the particles.
+
+.. math:: work_{t} = 10 * m_{t, in} * BWI * \left(\frac{1}{\sqrt{P_{t, prod, 80}}} - \frac{1}{\sqrt{P_{t, feed, 80}}}\right) 
+
+where :math:`work_{t}` is the work required to crush the particles at :math:`t` time, 10 is an empirical value and should not be changed, :math:`m_{t, in}` is the inlet mass flow rate
+at :math:`t` time, :math:`BWI` is the bond work index of particles, :math:`P_{t, prod, 80}` is production particle size with 80% passing the mesh at :math:`t` time, :math:`P_{t, feed, 80}` is 
+feed particle size with 80% passing the mesh at :math:`t` time.
+
+Expressions
+-----------
+
+Crusher includes two expressions to calculate the size of particles that has 80% passing the mesh for both feed and product particles.
+
+.. math:: P_{t, feed, 80} = \frac{S_{t, in, median}}{unit} * \left(-\log(1 - 0.8)\right)^{\frac{SW_{t, in}}{2}}
+
+where :math:`P_{t, feed, 80}` is the feed particle size that has 80% passing the mesh at :math:`t` time, :math:`\frac{S_{t, in, median}}{unit}` is the median particle size of input at :math:`t` time and unitless. The 
+default particle size is in micrometer. The :math:`SW_{t, in}` is the particle size width of input at :math:`t` time. 
+
+.. math:: P_{t, prod, 80} = \frac{S_{t, out, median}}{unit} * \left(-\log(1 - 0.8)\right)^{\frac{SW_{t, out}}{2}}
+
+where :math:`P_{t, prod, 80}` is the product particle size that has 80% passing the mesh at :math:`t` time, :math:`\frac{S_{t, out, median}}{unit}` is the median particle size of output at :math:`t` time and unitless. The 
+default particle size is in micrometer. The :math:`SW_{t, in}` is the particle size width of output at :math:`t` time. 
+
+Variables
+---------
+
+Crusher add the following additional variables beyond those created in property packages.
+
+================ ====== ================================================================================================
+Variable         Name   Notes
+================ ====== ================================================================================================
+:math:`work_{t}`  work
+================ ====== ================================================================================================
+
+"""
+
+from functools import partial
+from pyomo.environ import Var, log, Constraint, units as pyunits
+from pyomo.common.config import ConfigValue, ConfigDict, In
+from idaes.core import (
+    declare_process_block_class,
+    UnitModelBlockData,
+    useDefault,
+)
+from idaes.core.util.config import is_physical_parameter_block
+from idaes.core.util.tables import create_stream_table_dataframe
+import idaes.logger as idaeslog
+
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("CrushAndBreakageUnit")
```

Comment:
```suggestion
@declare_process_block_class("Crusher")
```
This is more of a question rather than a suggestion, but why choose the name CrushAndBreakgeUnit over something simpler like crusher?